src/task-implementation.rst

```diff
@@ -74,27 +74,31 @@ event handlers:
    cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" \
      "CUSTOM:data available for ${CYLC_TASK_CYCLE_POINT}"
 
-Custom severity messages and event handlers can be used to signal special
-events that are neither routine information or an error condition, such as
-production of a particular data file. Task output messages, used for triggering
-other tasks, can also be sent with custom severity if need be.
+These can be used to signal special events that are neither routine information
+nor an error condition, such as production of a particular data file (a "data
+availability" event).
 
-WARNING severity messages are printed to ``job.err``, logged by the
+"WARNING" severity messages are printed to ``job.err``, logged by the
 suite server program, and can be passed to *warning* event handlers:
 
 .. code-block:: bash
 
    cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" \
      "WARNING:Uh-oh, something's not right here."
 
-CRITICAL severity messages are printed to ``job.err``, logged by the
+"CRITICAL" severity messages are printed to ``job.err``, logged by the
 suite server program, and can be passed to *critical* event handlers:
 
 .. code-block:: bash
 
    cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" \
      "CRITICAL:ERROR occurred in process X!"
 
+Task jobs no longer (since Cylc 8) attempt to resend messages if the server
+cannot be reached. Send failures normally imply a network or Cylc configuration
+problem that will not recover by itself, in which case a series of messaging
+retries just holds up job completion unnecessarily. The server will recover
+task status by polling on job timeout, if a message does not get through.
```

Comment:
Perhaps add
... on job timeout (or earlier if execution polling is configured)

Reply:
Done.